### PR TITLE
Add Search bar and related info.

### PR DESCRIPTION
**Pull request by bnorton**

### CODE REVIEW

**Comment by bnorton** [comment]
[Modernize chmln.js + editor.js](https://trello.com/c/D8YUYqDp/3494-modernize-chmlnjs-editorjs)

File: docs/apis/search.md [hunk 1]
```diff
@@ -0,0 +1,712 @@
+# Search bar
+
+**Add a Search bar to your product that allows your Users to search your Product, Help center, and any custom content.**
+
+> **You have content within your product even if you don't think of it that way, any database records that your Users would want to find is a piece of content, akin to a Google search result. Send this content to Chameleon and we will surface it to your Users in the Chameleon search bar. The content can be scoped to a specific [Company](apis/company.md), be recommended when the Search bar is opened and can perform any action from clicking an item.**.
```
Comment (mamuh) [inline docs/apis/search.md:5]:
```suggestion
> **You have content within your product, even if you don't think of it that way. Any database records that your Users would want to find is a piece of content, akin to a Google search result. Send this content to Chameleon and we will surface it to your Users in the Chameleon search bar. The content can be scoped to a specific [Company](apis/company.md), be recommended when the Search bar is opened and can perform any action from clicking an item.**.
```

File: docs/apis/search.md [hunk 2]
```diff
@@ -0,0 +1,712 @@
...
+
+
+#### To get Search up and running, complete these items:
+
+1. Create a [Search bar](apis/search.md?id=schema), configure it with `placeholder` etc.
+2. Add your Help center as a [`SearchGroup`](apis/search.md?id=schema-search-group).
+3. Add the top-level navigation within your product; First download this CSV [template](apis/search.md?id=search-imports-template) or [example](apis/search.md?id=search-imports-example) then [Import](apis/search.md?id=schema-search-imports) it.
+4. [Optional] Add specific content from your database
+5. Publish your Search bar
+6. Visit your product where Chameleon is installed and hit `CMD` + `k` (or how you configured `key_meta` and `key_uid`).
```
Comment (mamuh) [inline docs/apis/search.md:22]:
```suggestion
6. Visit your product where Chameleon is installed and hit `CMD` + `k` for MacOs or 'Ctrl" + 'k' for Windows and Linux  (or how you configured `key_meta` and `key_uid`).
```

File: docs/apis/search.md [hunk 3]
```diff
@@ -0,0 +1,712 @@
...
+  }
+}
+```
+
+
+------
+
+## `SearchItem` Schema :id=schema-search-items
+
+A search item is a discrete unit of searchable content akin to a Google search result. You have content within your product even if you don't think of it that way, any database records that your Users would want to find is a piece of content. Sent us this content to surface it to your Users in the Chameleon search bar
```
Comment (shaggun) [inline docs/apis/search.md:262]:
```suggestion
A search item is a discrete unit of searchable content akin to a Google search result. You have content within your product even if you don't think of it that way, any database records that your Users would want to find is a piece of content. Send us this content to surface it to your Users in the Chameleon search bar
```

Comment (mamuh) [inline docs/apis/search.md:262]:
```suggestion
A search item is a discrete unit of searchable content akin to a Google search result. You have content within your product, even if you don't think of it that way. Any database records that your Users would want to find is a piece of content. Send us this content to surface it to your Users in the Chameleon search bar
```

**Review by mamuh — APPROVED** [review]

File: docs/apis/search.md [hunk 4]
```diff
@@ -0,0 +1,712 @@
...
+```javascript
+
+chmln.on('cmd:search', (opts: SearchOpts, ctx: Context) => {
+  /*
+  Optional:
+    - Chameleon will handle search for you
+    - If you need to add custom "Search" logic use this event to pass back the right items, two options:
+      - directly return an array<SearchItem>
+      - return a Promise that will resolve with the first argument being an array<SearchItem>.
+    - Best used when you already use Algolia or have a custo msearch endpoint internally
```
Comment (shaggun) [inline docs/apis/search.md:569]:
```suggestion
    - Best used when you already use Algolia or have a custom search endpoint internally
```

**Review by shaggun — APPROVED** [review]

**Comment by bnorton** [comment]
[HelpBar on private help centers](https://trello.com/c/5YKp5JQJ/4184-helpbar-on-private-help-centers)

**Pull request merged**